File: Examples/VisualBasic/ModifyingAndConvertingImages/ConvertEMFToWMF.vb
Imports Aspose.Imaging.ImageOptions

'
'This project uses Automatic Package Restore feature of NuGet to resolve Aspose.Imaging for .NET API reference 
'when the project is build. Please check https:// Docs.nuget.org/consume/nuget-faq for more information. 
'If you do not wish to use NuGet, you can manually download Aspose.Imaging for .NET API from http://www.aspose.com/downloads, 
'install it and then add its reference to this project. For any issues, questions or suggestions 
'please feel free to contact us using http://www.aspose.com/community/forums/default.aspx
'

Namespace Aspose.Imaging.Examples.VisualBasic.ModifyingAndConvertingImages
    Public Class ConvertEMFToWMF
        Public Shared Sub Run()
            ' The path to the documents directory.
            Dim dataDir As String = RunExamples.GetDataDir_ModifyingAndConvertingImages()

            ' Load an existing EMF file as Image.
            Using image__1 As Image = Image.Load(dataDir & Convert.ToString("Picture1.emf"))
                ' Call the Save method of Image class & Pass instance of WmfOptions class to Save method.
                image__1.Save(dataDir & Convert.ToString("ConvertEMFToWMF_out.wmf"), New WmfOptions())
            End Using
        End Sub
    End Class
End Namespace

File: Examples/CSharp/ModifyingAndConvertingImages/ConvertEMFToWMF.cs
// GIST-ID: d0acedb06fa0c40dc7da996a1292d818
using Aspose.Imaging.ImageOptions;

/*
This project uses the Automatic Package Restore feature of NuGet to resolve Aspose.Imaging for .NET API references
when the project is built. Please check https://learn.microsoft.com/en-us/nuget/resources/nuget-faq for more information.
If you do not wish to use NuGet, you can manually download Aspose.Imaging for .NET API from https://releases.aspose.com/,
install it, and then add its reference to this project. For any issues, questions, or suggestions,
please feel free to contact us using https://forum.aspose.com/
*/

namespace Aspose.Imaging.Examples.CSharp.ModifyingAndConvertingImages
{
    public class ConvertEMFToWMF
    {
        public static void Run()
        {
            // The path to the documents directory.
            string dataDir = RunExamples.GetDataDir_ModifyingAndConvertingImages();

            // Load an existing EMF file as Image.
            using (Image image = Image.Load(dataDir + "Picture1.emf"))
            {
                // Call the Save method of the Image class and pass an instance of WmfOptions to the Save method.
                var ops = image.GetDefaultOptions(null);
                var emfRasterization = ops.VectorRasterizationOptions as EmfRasterizationOptions;

                //EmfRasterizationOptions emfRasterization = new EmfRasterizationOptions();
                emfRasterization.BackgroundColor = Color.Yellow;
                emfRasterization.PageWidth = 100;
                emfRasterization.PageHeight = 100;
                emfRasterization.BorderX = 5;
                emfRasterization.BorderY = 10;
                
                image.Save(dataDir + "ConvertEMFToWMF_out.wmf", ops);
            }
        }
    }
}